frontend/public/components/cloud-services/install-plan.tsx

```diff
@@ -105,7 +96,77 @@ export class InstallPlanDetails extends React.Component<InstallPlanDetailsProps,
           </div>
         </div>
       </div>
-    </div>;
+    </div>
+  </React.Fragment>;
+};
+
+export class InstallPlanPreview extends React.Component<InstallPlanPreviewProps, InstallPlanPreviewState> {
+  constructor(props) {
+    super(props);
+    this.state = {needsApproval: this.props.obj.spec.approval === InstallPlanApproval.Manual && this.props.obj.spec.approved === false};
+  }
+
+  render() {
+    const {obj} = this.props;
+
+    const plan = _.get(obj.status, 'plan') || [];
+    const stepsByCSV = plan.reduce((acc, step) => acc.update(step.resolving, [], steps => steps.concat([step])), ImmutableMap<string, Step[]>()).toArray();
+
+    const approve = () => k8sUpdate(InstallPlanModel, {...obj, spec: {...obj.spec, approved: true}})
+      .then(() => this.setState({needsApproval: false}))
+      .catch((error) => this.setState({error}));
+
+    const stepStatus = (status: Step['status']) => <React.Fragment>
+      {status === 'Present' && <i className="fa fa-check-circle co-icon-space-r" style={{color: '#2ec98e'}} />}
```

Comment:
aria-hidden="true" here and just below

Avoid inline styles. Would be good to use one of the colors from the Patternfly palette:

https://www.patternfly.org/styles/color-palette/

We recently switched from this color to `$color-pf-green-400` in other places for legibility. See #123